run/john.conf

```diff
@@ -3497,6 +3497,40 @@ void restore()
 	}
 }
 
+# generate all possible wps pins
+[List.External:wpspin]
+int pin;
+void init() {
+	pin = 0;
+}
+void generate() {
+	if(pin > 9999999) {
+		word = 0;
+		return;
+	}
+	int i, p;
+	i = 0;
+	while(i < 8) word[i++] = '0';
+	word[8] = 0;
+	p = pin;
+	i = 6;
+	while(p) {
+		word[i] = '0' + p % 10;
+		p /= 10;
+		--i;
+	}
+	p = pin;
```

Comment:
tried `while(i >= 0) word[i--] = '0';` here instead of the explicit init on line 3512/13; though i couldn't reliably test for speed differences, sometimes old variant was faster, sometimes new one.